cps/editpage.py

```diff
@@ -0,0 +1,105 @@
+import os
+import flask
+from flask import Flask, abort, request
+from functools import wraps
+from pathlib import Path
+from flask_login import current_user, login_required
+
+from .render_template import render_title_template
+from . import logger, config, ub
+from .constants import CONFIG_DIR as _CONFIG_DIR
+
+log = logger.create()
+
+editpage = flask.Blueprint('editpage', __name__)
+
+def edit_required(f):
+    @wraps(f)
+    def inner(*args, **kwargs):
+        if current_user.role_edit() or current_user.role_admin():
+            return f(*args, **kwargs)
+        abort(403)
+
+    return inner
+
+def _get_checkbox(dictionary, field, default):
+    new_value = dictionary.get(field, default)
+    convertor = lambda y: y == "on"
+    new_value = convertor(new_value)
+
+    return new_value
+
+@editpage.route("/admin/page/<string:file>", methods=["GET", "POST"])
+@login_required
+@edit_required
+def edit_page(file):
+    doc = ""
+    title = ""
+    name = ""
+    icon = "file"
+    is_enabled = True
+    order = 0
+    position = "0"
+
+    page = ub.session.query(ub.Page).filter(ub.Page.id == file).first()
+    if page:
+        title = page.title
+        name = page.name
+        icon = page.icon
+        is_enabled = page.is_enabled
+        order = page.order
+        position = page.position
```

Comment:
```suggestion
        position = page.position
    except NotFound:
        if file != "new":
            abort(404)
```

Reply:
and import `from werkzeug.exceptions import NotFound`

Reply:
Done but changed NotFound to AttributeError